### PR TITLE
Add script to combine markdown files into a single page

### PR DESCRIPTION
**Pull request by MartinMinkov**
# Description:

This pull request introduces a new JavaScript script and a GitHub Actions workflow. The goal of the script is to combine all the markdown files in specified directories and their subdirectories into a single `.mdx` file. While combining markdown content, we remove any internal links so as not to break the site's build process. Instead, we replace the internal link with `/` to link to the main domain.

The script also supports excluding specific directories from the process.

The added GitHub Actions workflow runs the JS script in a CI environment, ensuring the combined markdown file is always up-to-date. It only runs on merged PRs that target the main. After a PR lands in `main`, the CI job will run and push an updated `onepage` to the repo.

New page is available on the preview deployment under `/onepage`

# Discussion:
The file `onepage.mdx` currently contains approximately `320,082` words. Based on the [guide for estimating token count from word count](https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them), the estimated token count is around `426,776` tokens. This may exceed the budget for a lower-cost model. What sections should be removed to reduce the token count? Do note, the filtering of content is based on files or directories, it's not granular enough to specify sections in a markdown file.

Please note that the `snarkyjs-api-reference` is not included in the markdown generation.

### CODE REVIEW

**Comment by barriebyron** [comment]
@MartinMinkov  is there something here for my language, word, or other editorial attention? 
I'd like to jump in and contribute, but I'm not clear on what expectations are. 

Is this PR for the AI initiative? Can we link to an issue? or provide context?

**Comment by MartinMinkov** [comment]
> Is this PR for the AI initiative? Can we link to an issue? or provide context?

Sorry about that! I linked it on Zenhub, but that doesn't seem to be the most reliable way to connect issues and PRs :grimacing: . I connected the original issue to this PR, which should be viewable in the "Development" section on the right of the page. This is the original issue here as well: https://github.com/o1-labs/docs2/issues/321

> @MartinMinkov is there something here for my language, word, or other editorial attention? I'd like to jump in and contribute, but I'm not clear on what expectations are.

To add some context to this work, the goal is to make a single page that is the concatenation of many different pages so that it can be fed into an AI like ChatGPT to give it context to answer questions about our docs. Given the discussion described in the PR, we could face a limit with the number of tokens that are present on this page. Which pages/directories do you seem most valuable to include so an AI bot can gain context on our docs? Currently, I am only specifying documents under `/zkapps` and leaving the rest of the docs as is for now.

**Comment by jasongitmail** [comment]
@barriebyron I wouldn't worry about any editing on this. It's just to make copying and pasting of zkApp and SnarkyJS-related docs easy for people using ChatGPT.

@MartinMinkov I see ChatGPT has enabled internet access for ChatGPT Plus users, so this might not be relevant very soon. We should move quickly with this and try to get it live Tues or Wed. They're slowing it rolling it out and we can ride the wave before that and have it guaranteed available for ETHTokyo

**Comment by MartinMinkov** [comment]
>  We should move quickly with this and try to get it live Tues or Wed.

Is there anything you need from me to finish this PR?

**Review by jasongitmail — COMMENTED** [review]
@MartinMinkov Do you know if there is any way to do this without the sidebar, header, footer? I notice that "cma+a cmd+c" copies those as well.

File: scripts/generate-one-page.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+const fs = require('fs');
+const path = require('path');
+
+// Define the base directory for the markdown files
+const docsDirectory = `${process.cwd()}/docs`;
+// List of directories containing markdown files to be combined
+const inputDirectories = ['zkapps'];
+// List of directories to be excluded from the combining process
+const excludedDirs = ['snarkyjs-reference'];
```
[the remark this answers is not in the document]
Comment (MartinMinkov) [reply scripts/generate-one-page.js:19]: Yeah, I have a discussion section in the PR description. I think the max token count for ChatGPT 4 is `64k` tokens, and we are already at `426,776` tokens given the documentation _excluding_ the API reference.

The token count was derived by counting the number of words inside the markdown file and then using the token guide to get an approximation. I think we need a more fine-grained strategy of what to include.

**Comment by MartinMinkov** [comment]
> Do you know if there is any way to do this without the sidebar, header, footer? I notice that "cma+a cmd+c" copies those as well.

I understand the appeal of implementing an easy copy-pasting feature via shortcuts. However, doing so would involve "ejecting" the default Docusaurus code to render pages and adding special conditions to exclude the TOC, header, and footer on this page. This approach would introduce a significant amount of technical debt, which might not be the best trade-off for simpler copy-pasting.

Alternatively, you can quickly copy the content by clicking the "Edit this page" button, focusing on the editor content in GitHub, and using Ctrl-A + Ctrl-C that way. This method should make the process more convenient without adding extra complexity to the codebase.

WDYT?

**Pull request closed without merging**